cmd/etcd-launcher/main.go

```diff
@@ -37,45 +39,44 @@ type envConfig struct {
 }
 
 type etcdCluster struct {
-	config      *envConfig
+	config      *config
 	client      *clientv3.Client
 	localClient *clientv3.Client
 }
 
 func main() {
 	// normal workflow, we don't do migrations anymore
 	e := &etcdCluster{}
-	err := e.getConfigFromEnv()
+	err := e.parseConfigFlags()
 	if err != nil {
 		log.Fatalf("failed to get launcher configuration: %v", err)
 	}
 
+	logOpts := kubermaticlog.NewDefaultOptions()
+	rawLog := kubermaticlog.New(logOpts.Debug, logOpts.Format)
+	log := rawLog.Sugar()
+	log = log.With("member-name", e.config.podName)
+
 	initialMembers := initialMemberList(e.config.clusterSize, e.config.namespace)
-	// not required, will leave it for now.
-	os.Setenv(initialStateEnvName, e.config.initialState)
-	os.Setenv(initialClusterEnvName, strings.Join(initialMembers, ","))
 
-	log.Print("initializing etcd..")
-	log.Printf("initial-state: %s", os.Getenv(initialStateEnvName))
-	log.Printf("initial-cluster: %s", os.Getenv(initialClusterEnvName))
+	log.Info("initializing etcd..")
+	log.Infof("initial-state: %s", e.config.initialState)
+	log.Infof("initial-cluster: %s", strings.Join(initialMembers, ","))
 
+	if _, err := os.Stat(etcdCommandPath); os.IsNotExist(err) {
+		log.Fatalf("can't find etcd command [%s]: %v", etcdCommandPath, err)
```

Comment:
Please follow the pattern that we use regaring contextual logs:  
`log.Fatalw("can't find command","command-path", etcdCommandPath, zap.Error(err))`